artdb/src/base/mod.rs

```diff
@@ -1,4 +1,24 @@
-pub mod storage_engine;
-pub mod page;
 pub mod buffer_pool;
-pub mod record;
\ No newline at end of file
+pub mod page;
+pub mod record;
+pub mod storage_engine;
+
+pub mod error {
```

Comment:
This is that "shared error" enum. It's a practice used quite a few projects (here's [bincode's](https://docs.rs/bincode/latest/bincode/enum.ErrorKind.html) for example)). The more advanced form does this thing where it pushes the kinds of errors into a separate enum (because then you can put it on the heap) but that's more down the road of stuff.

Tbh, [`thiserror`](https://docs.rs/thiserror/latest/thiserror/) makes this a _lot_ easier to do (and build). You wouldn't have to define the `From` trait by hand and it implements [`Error`](https://doc.rust-lang.org/std/error/trait.Error.html) for free.